Assets/Scripts/PowerUps/AbstractPowerUp.cs

```diff
@@ -0,0 +1,10 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class AbstractPowerUp
+{
+    public float modifier;
+    public abstract Vector3[] applyPowerUp(Vector3[] direction);
```

Comment:
Only BulletPowerUps should have this signature as their applyPowerUp. If this limits implementation, perhaps we should remove this base class.